### PR TITLE
Discover executables from gem, make tests pass on windows

### PR DESCRIPTION
**Pull request by ksubrama**

### CODE REVIEW

**Comment by ksubrama** [comment]
/cc @danielsdeleo @jaym

Functionally, nothing much changes other than cleaner log messages on the source and target.

File: lib/appbundler/app.rb [hunk 1]
```diff
@@ -128,25 +128,9 @@ def load_statement_for(bin_file)
...
-
-    def relative_app_lib_dir
-      lib_dir_pathname = Pathname.new(app_lib_dir)
-      bindir_pathname = Pathname.new(target_bin_dir)
-      lib_dir_pathname.relative_path_from(bindir_pathname).to_s
-    end
-
-    def app_lib_dir
-      File.join(app_root, "lib")
+      spec = Gem::Specification.find_by_name(app_spec.name, app_spec.version)
```
Comment (jaym) [inline lib/appbundler/app.rb]: Is app_spec a Gem::Specification?

Comment (ksubrama) [inline lib/appbundler/app.rb]: It's a Bundler::LazySpecification

Comment (danielsdeleo) [inline lib/appbundler/app.rb]: Can we add a check somewhere for the gem being installed (e.g., in the cli.rb) or rescue and print a nice error message here?

**Comment by ksubrama** [comment]
@danielsdeleo I updated it to print errors a bit earlier.

**Comment by danielsdeleo** [comment]
:+1:

**Pull request merged**